CMakeLists.txt

```diff
@@ -72,69 +90,156 @@ if (NOT HAVE_CXX_ATOMICS_WITHOUT_LIB
+            set(${version} "0x${ver}")
+        endif()
+    endmacro()
+
+    get_WIN32_WINNT(WINNT_VERSION)
+    message(STATUS "Windows NT version detection: ${WINNT_VERSION}")
+    if(NOT MSVC)
+        add_definitions(-D_WIN32_WINNT=${WINNT_VERSION} -DWINVER=${WINNT_VERSION})
     endif()
-    add_definitions(-DWINVER=${WINVER} -D_WIN32_WINNT=${WINVER})
+
+    set(CMAKE_DEBUG_POSTFIX "")
 endif()
 
 # Dependencies
 list (APPEND CMAKE_MODULE_PATH "${PROJECT_SOURCE_DIR}/cmake")
 if (NOT MSVC)
-    set (THREADS_PREFER_PTHREAD_FLAG TRUE)
-    find_package (Threads)
-    find_package (PkgConfig REQUIRED)
-    pkg_search_module (GnuTLS REQUIRED IMPORTED_TARGET gnutls)
-    pkg_search_module (Nettle REQUIRED IMPORTED_TARGET nettle)
-    check_include_file_cxx(msgpack.hpp HAVE_MSGPACKCXX)
-    if (NOT HAVE_MSGPACKCXX)
-        find_package(msgpack QUIET CONFIG NAMES msgpack msgpackc-cxx)
-        if (NOT msgpack_FOUND)
-            find_package(msgpack REQUIRED CONFIG NAMES msgpack-cxx)
-            set(MSGPACK_TARGET "msgpack-cxx")
-        else()
-            set(MSGPACK_TARGET "msgpackc-cxx")
-        endif()
-    endif()
-    if (OPENDHT_TOOLS)
-        find_package (Readline 6 REQUIRED)
-        set(readline_LIBRARIES ${READLINE_LIBRARIES})
-    endif ()
-    pkg_search_module(argon2 REQUIRED IMPORTED_TARGET libargon2)
-    set(argon2_lib ", libargon2")
-    pkg_search_module(Jsoncpp IMPORTED_TARGET jsoncpp)
-    if (Jsoncpp_FOUND)
+    if (WIN32 AND MINGW) # MSYS2
+        set (THREADS_PREFER_PTHREAD_FLAG TRUE)
+        find_package (Threads)
+        set(CMAKE_MAP_IMPORTED_CONFIG_RELWITHDEBINFO RELEASE)
+
+        # Workaround for msgpack-cxx
+        set(FETCHCONTENT_QUIET FALSE)
+        FetchContent_Declare(msgpack-c
+            GIT_REPOSITORY "https://github.com/msgpack/msgpack-c.git"
+            GIT_TAG        "cpp-7.0.0"
+            FIND_PACKAGE_ARGS NAMES msgpack-c 
+        )
+        set(MSGPACK_USE_BOOST OFF CACHE INTERNAL "Workaround for Boost")
+        FetchContent_MakeAvailable(msgpack-c)
+        
+        find_package(msgpack-c CONFIG REQUIRED)
+        include_directories(${MSGPACK_INCLUDEDIR} ${msgpack-c_SOURCE_DIR}/include)
+        
+        find_path(ARGON2_INCLUDE_DIR argon2.h REQUIRED)
+        include_directories(${ARGON2_INCLUDE_DIR})
+        
+        find_package(GnuTLS REQUIRED)
+        include_directories(${GNUTLS_INCLUDE_DIR})
+        
+        set(argon2_lib ", libargon2")
+        
+        find_package(jsoncpp CONFIG REQUIRED)
+        if (BUILD_SHARED_LIBS)
+            get_target_property(JSONCPP_INCLUDE_DIRS jsoncpp_lib INTERFACE_INCLUDE_DIRECTORIES)
+        else ()
+            get_target_property(JSONCPP_INCLUDE_DIRS jsoncpp_static INTERFACE_INCLUDE_DIRECTORIES)
+        endif ()
+        include_directories(${JSONCPP_INCLUDE_DIRS})
+        
+        find_package(fmt CONFIG REQUIRED)
+        include_directories(${FMT_INCLUDEDIR})
+        
```

Comment:
this new logic would prevent using pkgconfig with external dependencies with MINGW.
Also, if we have support for using FetchContent_Declare, it could be used on other platforms.

* don't prevent using installed dependencies on mingw
* FetchContent_Declare should be behind an option (default off)

Reply:
> Some libraries don't provide a CMake package, so we have to use pkg-config for these, at least on UNIX/MinGW.

@aberaud agreed. 

Regarding `FetchContent`: the only reason I kept it is the absence of a version for `MinGW64` in the MSYS2 repositories, so we need to copy, etc., and the package may be ported in the future. To avoid keeping on the machine the second MSYS2 environment for `clang`, I have decided to keep `FetchContent`, as `msgpack` is a header-only library. May we keep this fetcher for now, only for MSYS2, and adjust the other libraries for the traditional import tracks? For this PR, it would be better as it is more oriented to the removal of `WIN32_DEP_DIR`. and unification of Windows builds.

Reply:
@aberaud, the build pipeline logic seems now working. Would you like to have a look?